br/pkg/utils/suspend_importing.go

```diff
@@ -0,0 +1,136 @@
+package utils
+
+import (
+	"context"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/import_sstpb"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	berrors "github.com/pingcap/tidb/br/pkg/errors"
+	"github.com/pingcap/tidb/br/pkg/logutil"
+	"google.golang.org/grpc"
+)
+
+const (
+	DenyLightningUpdateFrequency = 5
+)
+
+func (mgr *StoreManager) GetAllStores(ctx context.Context) ([]*metapb.Store, error) {
+	return mgr.PDClient().GetAllStores(ctx)
+}
+
+func (mgr *StoreManager) GetDenyLightningClient(ctx context.Context, storeID uint64) (SuspendLightningClient, error) {
+	var cli import_sstpb.ImportSSTClient
+	err := mgr.WithConn(ctx, storeID, func(cc *grpc.ClientConn) {
+		cli = import_sstpb.NewImportSSTClient(cc)
+	})
+	if err != nil {
+		return nil, err
+	}
+	return cli, nil
+}
+
+type SuspendImportingEnv interface {
+	GetAllStores(ctx context.Context) ([]*metapb.Store, error)
+	GetDenyLightningClient(ctx context.Context, storeID uint64) (SuspendLightningClient, error)
+}
+
+type SuspendLightningClient interface {
```

Comment:
`SuspendLightningClient` is not proper because it's not suspend lightning only. also include br restore. so it's better to be `SuspendImportingClient` or `SuspendIngestClient`